core/drivers/scmi-msg/perf_domain.c

```diff
@@ -0,0 +1,402 @@
+/* SPDX-License-Identifier: BSD-3-Clause */
+/*
+ * Copyright (c) 2015-2020, Arm Limited and Contributors. All rights reserved.
+ * Copyright (c) 2021, Linaro Limited
+ * Copyright (c) 2024, STMicroelectronics
+ */
+#include <assert.h>
+#include <confine_array_index.h>
+#include <drivers/scmi-msg.h>
+#include <drivers/scmi.h>
+#include <mm/core_memprot.h>
+#include <string.h>
+#include <util.h>
+
+#include "common.h"
+#include "perf_domain.h"
+
+#define VERBOSE_MSG(...)		FMSG(__VA_ARGS__)
+
+static bool message_id_is_supported(unsigned int message_id);
+
+/* Weak handlers platform shall override on purpose */
+size_t __weak plat_scmi_perf_count(unsigned int channel_id __unused)
+{
+	return 0;
+}
+
+void __weak *plat_scmi_perf_statistics_buf(unsigned int channel_id __unused,
+					   size_t *stats_len)
+{
+	*stats_len = 0;
+
+	return NULL;
+}
+
+const char __weak *plat_scmi_perf_domain_name(unsigned int channel_id __unused,
+					      unsigned int domain_id __unused)
+{
+	return NULL;
+}
+
+int32_t __weak plat_scmi_perf_sustained_freq(unsigned int channel_id __unused,
+					     unsigned int domain_id __unused,
+					     unsigned int *freq __unused)
+{
+	return SCMI_NOT_SUPPORTED;
+}
+
+int32_t __weak plat_scmi_perf_levels_array(unsigned int channel_id __unused,
+					   unsigned int domain_id __unused,
+					   size_t start_index __unused,
+					   unsigned int *elt __unused,
+					   size_t *nb_elts __unused)
+{
+	return SCMI_NOT_SUPPORTED;
+}
+
+int32_t __weak plat_scmi_perf_level_latency_us(unsigned int channel_id __unused,
+					       unsigned int domain_id __unused,
+					       unsigned int level __unused,
+					       unsigned int *latency __unused)
+{
+	return SCMI_NOT_SUPPORTED;
+}
+
+int32_t __weak plat_scmi_perf_level_power_cost(unsigned int channel_id __unused,
+					       unsigned int domain_id __unused,
+					       unsigned int level __unused,
+					       unsigned int *cost __unused)
```

Comment:
I think this default implement should report `*cost = 0` ("power cost is not reported by the platform" as per the SCMI specification) and return with `SCMI_SUCCESS`.